src/main/java/ch/njol/skript/expressions/ExprInventorySlot.java

```diff
@@ -43,76 +43,77 @@
 
 @Name("Inventory Slot")
 @Description({"Represents a slot in an inventory. It can be used to change the item in an inventory too."})
-@Examples({"if slot 0 of player is air:",
+@Examples({
+	"if slot 0 of player is air:",
 	"\tset slot 0 of player to 2 stones",
 	"\tremove 1 stone from slot 0 of player",
-	"\tadd 2 stones to slot 0 of player",
-	"\tclear slot 1 of player"})
-@Since("2.2-dev24")
+	"\tadd 2 stones to slot 0 of inventory of all players",
+	"\tclear slot 1 of player"
+})
+@Since("2.2-dev24, INSERT VERSION (multiple inventories)")
 public class ExprInventorySlot extends SimpleExpression<Slot> {
-	
+
 	static {
 		Skript.registerExpression(ExprInventorySlot.class, Slot.class, ExpressionType.COMBINED,
-				"[the] slot[s] %numbers% of %inventory%", "%inventory%'[s] slot[s] %numbers%");
+			"[the] slot[s] %numbers% of %inventories%", "%inventories%'[s] slot[s] %numbers%");
 	}
 
 	@SuppressWarnings("null")
 	private Expression<Number> slots;
 	@SuppressWarnings("null")
-	private Expression<Inventory> invis;
-	
+	private Expression<Inventory> inventories;
+
 	@SuppressWarnings({"null", "unchecked"})
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		if (matchedPattern == 0){
-			 slots = (Expression<Number>) exprs[0];
-			 invis = (Expression<Inventory>) exprs[1];
+			slots = (Expression<Number>) exprs[0];
+			inventories = (Expression<Inventory>) exprs[1];
 		} else {
-			 slots = (Expression<Number>) exprs[1];
-			 invis = (Expression<Inventory>) exprs[0];			
+			slots = (Expression<Number>) exprs[1];
+			inventories = (Expression<Inventory>) exprs[0];
 		}
 		return true;
 	}
 
 	@Override
 	@Nullable
 	protected Slot[] get(Event event) {
-		Inventory invi = invis.getSingle(event);
-		if (invi == null)
-			return null;
-		
 		List<Slot> inventorySlots = new ArrayList<>();
-		for (Number slot : slots.getArray(event)) {
-			if (slot.intValue() >= 0 && slot.intValue() < invi.getSize()) {
+		for (Inventory inventory : inventories.getArray(event)) {
+			if (inventory == null)
+				continue;
+			for (Number slot : slots.getArray(event)) {
```

Comment:
You should cache the value of `slots.getArray(event)` outside of the loops then loop that instead

Reply:
Does this applies to inventories.getArray(event), or it is fine as is?

Reply:
No, that should be fine because it isn't getting called multiple times, unlike `slots.getArray`